akka-stream-tests/src/test/scala/akka/stream/impl/ChainedBufferSpec.scala

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (C) 2022 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream.impl
+
+import akka.stream.testkit.StreamSpec
+import scala.annotation.tailrec
+import org.scalatest.compatible.Assertion
+
+class ChainedBufferSpec extends StreamSpec {
+  val doNothing: Any => Unit = _ => ()
+
+  val random = scala.util.Random.nextInt(16)
+  val headCapacity = random / 4 + 1
+  val tailCapacity = random % 4 + 1
```

Comment:
Since I think the case is reasonably strong for not having the extra buffers (though the chained buffer might be useful more generally?), I'm going to wait on this bit of feedback.